lib/velveteen/worker.rb

```diff
@@ -43,8 +43,29 @@ def logger
     end
 
     def publish(payload, options = {})
+      validate_published_message!(payload, options[:routing_key])
       options[:headers] = build_headers(options)
-      Config.exchange.publish(payload, options)
+      Config.exchange.publish(payload.to_json, options)
+    end
+
+    def validate_published_message!(message_to_publish, publish_routing_key)
+      errors = JSON::Validator.fully_validate(
+        schema_from_routing_key(publish_routing_key),
+        message_to_publish
+      )
+
+      if errors.any?
+        raise InvalidMessage, errors.join("\n")
+      end
+    end
+
+    def schema_from_routing_key(routing_key)
+      schema_file_path = file_path_from_routing_key(routing_key)
+      File.expand_path(schema_file_path, Config.schema_directory)
+    end
+
+    def file_path_from_routing_key(routing_key)
+      routing_key.tr(".", "-") + ".json"
```

Comment:
What do you think about using the routing key verbatim in the file name? E.g. `velveteen.fail.randomly.json`. I think I'd appreciate that slight reduction in overhead when wading through a directory full of schema files.

Reply:
hmmm, yeah into it!